src/main/java/net/minecraftforge/client/event/OverlayGlintEvent.java

```diff
@@ -0,0 +1,87 @@
+package net.minecraftforge.client.event;
+
+import net.minecraft.client.model.ModelBase;
+import net.minecraft.client.resources.model.IBakedModel;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+/**
+ * Events for hooking into the Glint effect Rendering<br>
+ * Will allow for adding effects to items, creating custom overlays, modifying vanilla
+ * colors, and canceling of glint overlays. 
+ * @author RANKSHANK
+ */
+public class OverlayGlintEvent<T> extends Event
+{
```

Comment:
Generics can screw things up, make sure to mark this as an abstract class so that modders do not instantiate it itself.